roles/addons/diskless/files/disklessset.py

```diff
@@ -327,7 +327,7 @@ def generate_ipxe_boot_file(image_type, image_name, image_kernel, image_initramf
                 os.system('dnf install -y iproute procps-ng openssh-server NetworkManager  --installroot=/mnt/ --exclude glibc-all-langpacks --exclude cracklib-dicts --exclude grubby --exclude libxkbcommon --exclude pinentry --exclude python3-unbound --exclude unbound-libs --exclude xkeyboard-config --exclude trousers --exclude diffutils --exclude gnupg2-smime --exclude openssl-pkcs11 --exclude rpm-plugin-systemd-inhibit --exclude shared-mime-info --exclude glibc-langpack-* --setopt=module_platform_id=platform:el8 --nobest')
                 os.system('dnf install -y dnf  --installroot=/mnt/ --exclude glibc-all-langpacks --exclude cracklib-dicts --exclude grubby --exclude libxkbcommon --exclude pinentry --exclude python3-unbound --exclude unbound-libs --exclude xkeyboard-config --exclude trousers --exclude diffutils --exclude gnupg2-smime --exclude openssl-pkcs11 --exclude rpm-plugin-systemd-inhibit --exclude shared-mime-info --exclude glibc-langpack-* --setopt=module_platform_id=platform:el8 --nobest')
             elif selected_livenet_type == '1':
-                os.system('dnf groupinstall -y "core"  --setopt=module_platform_id=platform:el8 --installroot=/mnt')
+                os.system('dnf install -y @core kernel --setopt=module_platform_id=platform:el8 --installroot=/mnt')
```

Comment:
The file /lib/modules/$(uname -r)/modules.builtin is installed by kernel-core, which should be enough to solve the issue.

Reply:
Not sure kernel-core is enough, must be tested, but i do agree with the second remark, can you test it with following line ?

`os.system('dnf install -y @core kernel-core-{} --setopt=module_platform_id=platform:el8 --installroot=/mnt'.format(kernel_list[int(selected_kernel)].strip('vmlinuz-')))`

Also looks like an heavy package, sizes may be updated for images 2 and 3 if this modification is required

Reply:
Indeed the core group already installs kernel-core as a dependency, but its post-install transaction fails:

```
  Running scriptlet: kernel-core-4.18.0-147.el8.x86_64
/usr/lib/kernel/install.d/50-dracut.install: line 41: /proc/cmdline: No such file or directory
dracut: No '/dev/log' or 'logger' included for syslog logging
libkmod: kmod_module_new_from_loaded: could not open /proc/modules: No such file or directory
dracut-install: Could not get list of loaded modules: Unknown error -2. Switching to non-hostonly mode.
grep: /proc/cpuinfo: No such file or directory
grep: /proc/cpuinfo: No such file or directory
grep: /proc/cpuinfo: No such file or directory
grep: /proc/cpuinfo: No such file or directory
dracut: Turning off host-only mode: '/sys' is not mounted!
dracut: Turning off host-only mode: '/proc' is not mounted!
dracut: Turning off host-only mode: '/run' is not mounted!
dracut: Turning off host-only mode: '/dev' is not mounted!
/bin/sed: can't read /proc/cpuinfo: No such file or directory
/bin/sed: can't read /proc/cpuinfo: No such file or directory
/bin/sed: can't read /proc/cpuinfo: No such file or directory
/bin/sed: can't read /proc/cpuinfo: No such file or directory
/usr/lib/kernel/install.d/51-dracut-rescue.install: line 53: /proc/cmdline: No such file or directory
Could not determine the kernel command line parameters.
Please specify the kernel command line in /etc/kernel/cmdline!
warning: %posttrans(kernel-core-4.18.0-147.el8.x86_64) scriptlet failed, exit status 1

Error in POSTTRANS scriptlet in rpm package kernel-core
```

When I mount proc in the installroot directory, the post-install completes and the file /lib/modules/4.18.0-147.el8.x86_64/modules.builtin  exists in the rootfs.img. However it does not match the booted kernel (the kernel used to create the image is vmlinuz-4.18.0-147.3.1.el8_1.x86_64).

Reply:
When using CentOS 8.1 ISO as repository (as I did above), the kernel-core is installed. Doing further tests, kernel-core is not installed anymore with later releases.

After moving to http://mirror.centos.org/centos/8.1.1911/BaseOS/x86_64/os/, the package is not installed.
Confirmed with http://mirror.centos.org/centos/8.2.2004/BaseOS/x86_64/kickstart/ (kickstart is a snapshot of os at GA/release time)

Reply:
> The file /lib/modules/$(uname -r)/modules.builtin is installed by kernel-core, which should be enough to solve the issue.

RPM kernel installs kernel-module and kernel-core.
Both are important.